core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -78,7 +78,11 @@ private[spark] class SparkDeploySchedulerBackend(
     val appUIAddress = sc.ui.map(_.appUIAddress).getOrElse("")
     val appDesc = new ApplicationDescription(sc.appName, maxCores, sc.executorMemory, command,
       appUIAddress, sc.eventLogDir)
-
+    appDesc.coreNumPerTask = conf.getInt("spark.task.cpus", 1)
```

Comment:
Eh...seems like I didn't make it clear. I mean in `SparkDeploySchedulerBackend` using a `val` to represent `conf.getInt("spark.task.cpus", 1)` is better. Like:

> val coreNumPerTask = conf.getInt("spark.task.cpus", 1)
if ( coreNumPerTask  < 1 ) {
   throw ....
} else {
  appDesc.....
}